lib/astutils.cpp

```diff
@@ -2676,8 +2676,13 @@ bool isVariableChanged(const Token *tok, int indirect, const Settings &settings,
             return false;
 
         const Token *ftok = tok2->astParent()->astOperand2();
-        if (astIsContainer(tok2->astParent()->astOperand1()) && vt && vt->container) {
-            const Library::Container* c = vt->container;
+        const Token* ctok = tok2->astParent();
+        if (Token::simpleMatch(ctok->astOperand1(), "."))
```

Comment:
hmm.. we know that `ctok->astOperand1()` is `tok2` thanks to the condition at line 2673.